pkg/apis/servicecatalog/plan_reference.go

```diff
@@ -105,10 +139,38 @@ func (pr PlanReference) GetPlanFilterFieldName() string {
 	return ""
 }
 
+// GetServiceClassFilterFieldName returns the appropriate field name for filtering
+// a list of service catalog classes by the PlanReference.
+func (pr PlanReference) GetServiceClassFilterFieldName() string {
+	if pr.ServiceClassExternalName != "" {
+		return "spec.externalName"
+	}
+
+	if pr.ServiceClassExternalID != "" {
+		return "spec.externalID"
+	}
+
+	return ""
+}
+
+// GetServicePlanFilterFieldName returns the appropriate field name for filtering
+// a list of service catalog plans by the PlanReference.
+func (pr PlanReference) GetServicePlanFilterFieldName() string {
+	if pr.ServicePlanExternalName != "" {
+		return "spec.externalName"
+	}
+
+	if pr.ServicePlanExternalID != "" {
+		return "spec.externalID"
+	}
+
+	return ""
+}
+
 // String representation of a PlanReference
 // Example: class_name/plan_name, class_id/plan_id
 func (pr PlanReference) String() string {
-	return fmt.Sprintf("%s/%s", pr.GetSpecifiedClass(), pr.GetSpecifiedPlan())
+	return fmt.Sprintf("%s/%s", pr.GetSpecifiedClusterServiceClass(), pr.GetSpecifiedClusterServicePlan())
```

Comment:
The intent of this function is that when we print a plan reference that it prints out the class and plan. Right now this won't print anything when the class/plan is namespaced scope.